crates/hir_ty/src/db.rs

```diff
@@ -128,6 +128,12 @@ pub trait HirDatabase: DefDatabase + Upcast<dyn DefDatabase> {
         krate: CrateId,
         env: chalk_ir::Environment<chalk::Interner>,
     ) -> chalk_ir::ProgramClauses<chalk::Interner>;
+
+    #[salsa::invoke(crate::lower::field_visibilities_query)]
+    fn field_visibilities(&self, var: VariantId) -> Arc<ArenaMap<LocalFieldId, Visibility>>;
+
+    #[salsa::invoke(crate::lower::fn_visibility_query)]
+    fn fn_visibility(&self, def: FunctionId) -> Visibility;
```

Comment:
can we move these to `DefDatabase`?

Reply:
I’ve also noticed this and already made the changes locally but I forgot to push it :)